### PR TITLE
Move cloud-init profile to a dedicated method

### PR DESCRIPTION
**Pull request by mpagot**
Ipaddr2 library move creation of cloud-init profile to its own method.
Add support for nginx installation on 12sp5 from an external repo

- Related ticket: TEAM-8721

# Verification run:

## 12sp5
http://openqaworker15.qa.suse.cz/tests/300942 :green_circle: Here where nginx is properly installed http://openqaworker15.qa.suse.cz/tests/300942#step/configure/206. Failure is due to missing socat, something addressed by PR https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20456

 http://openqaworker15.qa.suse.cz/tests/300981

 http://openqaworker15.qa.suse.cz/tests/300983

## 15sp5
http://openqaworker15.qa.suse.cz/tests/300944 :green_circle: 
 http://openqaworker15.qa.suse.cz/tests/300982 :green_circle: 
 http://openqaworker15.qa.suse.cz/tests/300984 :green_circle:

### CODE REVIEW

File: lib/sles4sap/ipaddr2.pm [hunk 1]
```diff
@@ -88,6 +89,79 @@ sub ipaddr2_azure_storage_account {
...
+    my $cloud_init_content = <<END;
+#cloud-config
+package_upgrade: false
+END
+    # 2. Install and configure nginx, two variants as
+    #    12sp5 does not have the nginx package
+    if ($args{external_repo}) {
+        $cloud_init_content .= <<END;
+runcmd:
+  - 'sudo zypper addrepo -G -t yum -c "http://nginx.org/packages/sles/$args{external_repo}"" nginx",
```
Comment (jankohoutek) [inline lib/sles4sap/ipaddr2.pm:134]: No, no, no, just NO!

File: t/22_ipaddr2.t [hunk 2]
```diff
@@ -35,82 +30,33 @@ subtest '[ipaddr2_infra_deploy] no BYOS' => sub {
...
-    $ipaddr2->redefine(az_vm_wait_running => sub { return; });
-
-    my $azcli = Test::MockModule->new('sles4sap::azure_cli', no_auto => 1);
-    $azcli->redefine(assert_script_run => sub { push @calls, ['azure_cli', $_[0]]; return; });
-    $azcli->redefine(script_output => sub { push @calls, ['azure_cli', $_[0]]; return 'Fermi'; });
-
-    ipaddr2_infra_deploy(region => 'Marconi', os => 'Meucci:gen2:ByoS', cloudinit => 0);
-
-    # push th elist of commands in another list, this one withour the source
+    # push the list of commands in another list, this one withour the source
```
Comment (lpalovsky) [inline t/22_ipaddr2.t:59]: s/withour/without/

File: t/22_ipaddr2.t [hunk 3]
```diff
@@ -160,9 +82,9 @@ subtest '[ipaddr2_infra_deploy] no cloud-init' => sub {
     $azcli->redefine(assert_script_run => sub { push @calls, ['azure_cli', $_[0]]; return; });
     $azcli->redefine(script_output => sub { push @calls, ['azure_cli', $_[0]]; return 'Fermi'; });
 
-    ipaddr2_infra_deploy(region => 'Marconi', os => 'Meucci:gen2:ByoS', cloudinit => 0);
+    ipaddr2_infra_deploy(region => 'Marconi', os => 'Meucci:gen2:ByoS');
 
-    # push th elist of commands in another list, this one withour the source
+    # push the list of commands in another list, this one withour the source
```
Comment (lpalovsky) [inline t/22_ipaddr2.t:87]: s/withour/without/

File: t/22_ipaddr2.t [hunk 4]
```diff
@@ -866,14 +825,70 @@ subtest '[ipaddr2_os_cloud_init_logs]' => sub {
...
+    $ipaddr2->redefine(upload_logs => sub { return '/Faggin'; });
+
+    ipaddr2_cloudinit_create(nginx_root => 'ABCD');
+
+    note("cloud_init_content:\n" .
+          "--------------------------\n" .
+          $cloud_init_content .
+          "\n--------------------------");
+
+    #like($cloud_init_content, qr/registercloudguest.*ABCD/, "cloud-init script registers");
```
Comment (lpalovsky) [inline t/22_ipaddr2.t:888]: Commented out code. Is it required?

**Review by lpalovsky — APPROVED** [review]
LGTM, there are some typos though.

File: t/22_ipaddr2.t (hunk 4 above)
Comment (mpagot) [reply t/22_ipaddr2.t:888]: o yes it is, or better to say the next `ok(1)` has to be converted to something more usefull

**Review by jankohoutek — APPROVED** [review]
LGTM

**Pull request merged**